bookshelf/src/main/webapp/base.jsp

```diff
@@ -34,9 +34,11 @@ Copyright 2015 Google Inc. All Rights Reserved.
         </div>
         <ul class="nav navbar-nav">
           <li><a href="/">Books</a></li>
-          <li><a href="/books/mine">My Books</a></li>
+          <div class="form-group ${isAuthConfigured ? '' : 'hidden'}">
```

Comment:
Ok, this and the next comment make sense. If you want to continue testing your version we can use that. My only suggestion would be do all the property checking in one place, such as the init function in ListBookServlet